packs/vsphere/actions/vm_hw_power.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import eventlet
+from pyVmomi import vim
+
+from vmwarelib import inventory
+from vmwarelib import checkinputs
+from vmwarelib.actions import BaseAction
+
+
+class VMApplyPowerState(BaseAction):
+
+    def run(self, vm_id, vm_name, power_onoff):
+        # check I have information to find a VM
+        checkinputs.one_of_two_strings(vm_id, vm_name, "ID or Name")
+        # convert ids to stubs
+        vm = inventory.get_virtualmachine(self.si_content,
+                                          moid=vm_id, name=vm_name)
+        if not vm:
+            raise Exception('Error: Unable to find VM')
```

Comment:
Yeah, I think it's fine since `ValueError` refers to function / method arguments (e.g. argument is of a right type but contains an invalid / unsupported value) and not the result.